lib/elixir_analyzer/exercise_test/common_checks/compiler_warnings.ex

```diff
@@ -5,28 +5,30 @@ defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.CompilerWarnings do
   alias ElixirAnalyzer.Constants
   alias ElixirAnalyzer.Comment
 
-  def run(code_path, _code_ast) do
-    import ExUnit.CaptureIO
-    Application.put_env(:elixir, :ansi_enabled, false)
+  def run(code_path) do
+    Logger.configure(level: :critical)
 
     warnings =
-      capture_io(:stderr, fn ->
-        try do
-          Code.compile_file(code_path)
-          |> Enum.each(fn {module, _binary} ->
+      case Kernel.ParallelCompiler.compile(code_path) do
```

Comment:
A big change is the format of warnings that went from

```
warning: Behaviour.defcallback/1 is deprecated. Use the @callback module attribute instead
  test_data/lasagna/deprecated_modules/lib/lasagna.ex:4: Lasagna
```
to
```
warning: Behaviour.defcallback/1 is deprecated. Use the @callback module attribute instead
  lib/lasagna.ex:4
```

First change: less information about the module/function with the warning, we now only have the line number. This is due to `Kernel.ParallelCompiler.compile` not actually giving that information, even though is is printed out in stdout. I could try to capture the IO output, but it makes other things more complicated, so I thought line number would be good enough.

Second change: the file path now starts at `lib/`. This is 100% my choice, because I've always disliked seeing the full path from the docker container, which is noise and implementation detail. Since we have to format warnings now, I thought I might as well clean them up.